### PR TITLE
bpo-33753: Refactor creating temporary files in test_fileinput.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue33753

### CODE REVIEW

File: Lib/test/test_fileinput.py [hunk 1]
```diff
@@ -33,20 +33,17 @@
...
-    for name in names:
-        if name:
-            safe_unlink(name)
+class BaseTests:
+    # Write a content (str or bytes) to temp file, and return the
+    # temp file's name.
+    count = 0
+    def writeTmp(self, content, *, mode='w'):  # opening in text mode is the default
+        self.count += 1
+        name = TESTFN + str(self.count)
```
Comment (vstinner) [inline Lib/test/test_fileinput.py]: Why not using tempfile.mktemp()? tempfile uses a temporary directory which is cleared sometimes at system boot. It prevents leaking temporary files if Python does crash.

**Review by vstinner — APPROVED** [review]
LGTM. addCleanup() is a good solution.

I asked a question, but it shouldn't be a blocker issue.

File: Lib/test/test_fileinput.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/test_fileinput.py]: Good idea!

**Review by vstinner — APPROVED** [review]
Ah, I proposed mktemp() but mkstemp() may be even better.

**Comment by serhiy-storchaka** [comment]
`mktemp()` is half-deprecated. THIS FUNCTION IS UNSAFE AND SHOULD NOT BE USED.

**Pull request merged**